src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Rename/RenamerProjectTreeActionHandler.cs

```diff
@@ -83,6 +83,7 @@ public override async Task RenameAsync(IProjectTreeActionHandlerContext context,
 
             // Get the list of possible actions to execute
             string oldName = Path.GetFileNameWithoutExtension(oldFilePath);
+            string newName = Path.GetFileNameWithoutExtension(newFileWithExtension);
```

Comment:
I think this will end up throwing before CPS has had a chance to verify the path characters, and you'll override their error message.

Reply:
I changed the order to get newName after CPSRenameAsync